python/cuml/common/device_selection.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import cuml
+import contextlib
+
+
+def set_global_device_type(device_type):
+    if (isinstance(device_type, str)):
+        device_type = device_type.lower()
+
+    if device_type not in ['cpu', 'gpu']:
```

Comment:
How would we feel about making this an enum with an appropriate string-to-enum conversion so the user never has to explicitly touch the enum themself?